### DISCUSSION TITLE
Use IDocumentStore.BulkInsert inside a TransactionScope?

### DISCUSSION THREAD

**Discussion by youqingz** [JasperFx/marten#2148] [Q&A]
I can use IDocumentSession to enlist an ambient transaction inside a TransactionScope, but I also would like to use IDocumentStore.BulkInsert for some of the operations for performance reasons, but seems like it is not supported at the moment?

**Comment by jeremydmiller** [comment]
That's correct. Not something that's ever come up before.

**Comment by jeremydmiller** [comment]
Correct. Not something that's ever come up. We take pull requests though!

**Reply by youqingz** [reply]
Thanks, I'll have a go at the PR.

**Reply by youqingz** [reply]
@jeremydmiller should I add a optional Transaction parameter and enlist it or Create a new set of methods called BulkInsertEnlistingTransaction?

**Reply by jeremydmiller** [reply]
I'd vote for an all new method that's semantically different. Sorry for the delay responding here.
